pkg/secrethub/internals/http/client.go

```diff
@@ -245,18 +245,40 @@ func (c *Client) DeleteRepo(namespace, repoName string) error {
 }
 
 // AuditRepo gets the audit events for a given repo.
-func (c *Client) AuditRepo(namespace, repoName string, subjectTypes api.AuditSubjectTypeList) ([]*api.Audit, error) {
-	out := []*api.Audit{}
-	rawURL := fmt.Sprintf(pathRepoEvents, c.base, namespace, repoName)
-	query := make(url.Values)
-	if len(subjectTypes) > 0 {
-		query.Set("subject_types", subjectTypes.Join(","))
+func (c *Client) AuditRepo(namespace, repoName string, subjectTypes api.AuditSubjectTypeList) (*Paginator, error) {
+	requestURL, err := url.Parse(fmt.Sprintf(pathRepoEvents, c.base, namespace, repoName))
+	if err != nil {
+		return nil, fmt.Errorf("invalid URL: %s", err)
 	}
-	if len(query) > 0 {
-		rawURL += "?" + query.Encode()
+
+	if len(subjectTypes) > 0 {
+		q := requestURL.Query()
+		q.Set("subject_types", subjectTypes.Join(","))
+		requestURL.RawQuery = q.Encode()
 	}
-	err := c.get(rawURL, true, &out)
-	return out, errio.Error(err)
+
+	return c.auditPage(*requestURL), nil
+}
+
+func (c *Client) auditPage(url url.URL) *Paginator {
```

Comment:
I'd love to move away from `url.Parse`. Instead, we could validate the URL (and return an error) when setting the base URL of the client.

Reply:
Now that I've said this, let's do this now, so that the signature of `client.Repos().EventIterator(..)` changes to only return a `Paginator` and no `error`.